lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -465,31 +495,73 @@ static IntegerAttr getI1IntegerAttr(MLIRContext *context, bool value) {
                           static_cast<int64_t>(value));
 }
 
-OpFoldResult AtenGtIntOp::fold(ArrayRef<Attribute> operands) {
-  auto lhs = operands[0].dyn_cast_or_null<IntegerAttr>();
-  auto rhs = operands[1].dyn_cast_or_null<IntegerAttr>();
-  if (lhs && rhs) {
-    return getI1IntegerAttr(getContext(), lhs.getValue().getSExtValue() >
-                                              rhs.getValue().getSExtValue());
-  }
-  return nullptr;
+using ConstantIntComparator = std::function<bool(int, int)>;
+template <typename OpTy>
+static OpFoldResult comparatorFoldHelper(OpTy op, bool equalIsTrue,
+                                         ConstantIntComparator comparator) {
```

Comment:
equalIsTrue is redundant with comparator. We should always have `equalIsTrue == comparator(0, 0)`.